src/odf/dataset/src/services/metadata_helpers.rs

```diff
@@ -195,6 +195,26 @@ async fn resolve_transform_inputs(
 ) -> Result<(), ValidateDatasetSnapshotError> {
     let mut missing_inputs = Vec::new();
 
+    let mut dataset_ref_vec = inputs
+        .iter()
+        .map(|i| i.dataset_ref.clone())
+        .collect::<Vec<_>>();
+    dataset_ref_vec.sort_unstable();
+    let duplicate_inputs: Vec<_> = dataset_ref_vec
+        .windows(2)
+        .filter(|&dataset_ref| (dataset_ref[0] == dataset_ref[1]))
+        .map(|dataset_ref| dataset_ref[0].clone())
+        .collect();
```

Comment:
We can have three identical inputs, and then windows by 2 won't catch it.

I would suggest a solution using `HashSet<t>`:
- go through all inputs
- try to insert into the set
- if it already exists, add to `duplicate_inputs`